src/app/workshops/workshops.component.ts

```diff
@@ -14,54 +14,53 @@ export class WorkshopsComponent {
-	
-	private globalConstants:GlobalConstantsRepository;
-	
-	@ViewChild(WorkshopsListComponent) workshopsListChildComp:WorkshopsListComponent;
-	@ViewChild(WorkshopFilterComponent) workshopsFilterChildComp:WorkshopFilterComponent;
-
-	constructor(
-        private globalConstantsRepository:GlobalConstantsRepository,
-        private route:ActivatedRoute,
-        private router:Router,
+    private readonly workshopsPerPage: number = 8;
+
+    private globalConstants: GlobalConstantsRepository;
+
+    @ViewChild(WorkshopsListComponent) workshopsListChildComp: WorkshopsListComponent;
+    @ViewChild(WorkshopFilterComponent) workshopsFilterChildComp: WorkshopFilterComponent;
+
+    constructor(
+        private globalConstantsRepository: GlobalConstantsRepository,
+        private route: ActivatedRoute,
+        private router: Router,
         title: Title,
         meta: Meta,
-        public gaService:GoogleAnalyticsService)
-	{
+        public gaService: GoogleAnalyticsService) {
         this.globalConstants = globalConstantsRepository;
         this.hideFilter = true;
 
         this.gaService.trackPageView('Workshops');
-        
+
         title.setTitle('List of Photography Workshops');
         meta.addTags([
             { name: 'twitter:title', content: 'List of Photography Workshops' },
             { property: 'og:title', content: 'List of Photography Workshops' },
-            { property: 'og:type', content: 'website'},
-            { property: 'og:site_name', content: 'The Pixelated Planet'},
-            { property: 'fb:app_id', content: '132676104124561'},
+            { property: 'og:type', content: 'website' },
+            { property: 'og:site_name', content: 'The Pixelated Planet' },
+            { property: 'fb:app_id', content: '132676104124561' },
             { name: 'description', content: 'Photography workshops list which can be filtered on by Location, Category, Dates and Price' },
             { property: 'og:description', content: 'Photography workshops list which can be filtered on by Location, Category, Dates and Price' },
             { name: 'twitter:description', content: 'Photography workshops list which can be filtered on by Location, Category, Dates and Price' },
             { property: 'og:image', content: 'http://pixelatedplanetcdn.azureedge.net/img/yosemite.jpg' },
             { name: 'twitter:image', content: 'http://pixelatedplanetcdn.azureedge.net/img/yosemite.jpg' },
             { property: 'og:url', content: `https://www.thepixelatedplanet.com${this.router.url}` },
             { name: 'twitter:site', content: `https://www.thepixelatedplanet.com${this.router.url}` },
-          ]);
-        
+        ]);
+
         router.events.subscribe(event => {
-            if(event instanceof NavigationEnd)
-                {
-                    this.updateUrl();
-                }
+            if (event instanceof NavigationEnd) {
+                this.setParameters();
```

Comment:
you added this back :D I thought the whole thing started with removing this?

Reply:
same as above. And this was an issue since we were getting extra calls because of this but now every call is make once only except the autocomplete which we anyway require.